sdk/core/core/src/az_span.c

```diff
@@ -584,7 +584,7 @@ AZ_NODISCARD static az_span _az_span_trim_side(az_span source, az_span_trim_side
   if (side == RIGHT)
   {
     increment = -1; // Set increment to be decremental for moving ptr
-    source_ptr += (source_size - 1); // Set initial position to the end
+    source_ptr += ((size_t)source_size - 1); // Set initial position to the end
```

Comment:
We don't cast to `size_t` anywhere else. Why was this necessary?

Reply:
There was warning on my dogfood MSVC.